src/horus_api.c

```diff
@@ -98,19 +138,42 @@ struct horus *horus_open (int mode, int Rs) {
         hstates->rx_bits_len = hstates->max_packet_len;
     }
 
-    if (mode == HORUS_MODE_BINARY) {
+    if (mode == HORUS_MODE_BINARY_V1) {
+        // Parameter setup for the Legacy Horus Binary Mode (22 byte frames, Golay encoding)
+
         hstates->mFSK = 4;
-        hstates->max_packet_len = HORUS_BINARY_NUM_BITS;
-        for (i=0; i<sizeof(uw_horus_binary); i++) {
-            hstates->uw[i] = 2*uw_horus_binary[i] - 1;
+        hstates->max_packet_len = HORUS_BINARY_V1_NUM_CODED_BITS;
+
+        // If baud rate not provided, use default
+        if (hstates->Rs == -1){
+            hstates->Rs = HORUS_BINARY_V1_DEFAULT_BAUD;
+        }
+
+        if (tx_tone_spacing == -1){
+            // No tone spacing provided. Disable mask estimation, and use the default tone spacing value as a dummy value.
+            tx_tone_spacing = HORUS_BINARY_V1_DEFAULT_TONE_SPACING;
+            mask = 0;
+        } else {
+            // Tone spacing provided, enable mask estimation.
+            mask = 1;
         }
-        hstates->uw_len = sizeof(uw_horus_binary);
-        hstates->uw_thresh = sizeof(uw_horus_binary) - 2; /* allow a few bit errors in UW detection */
+
+        for (i=0; i<sizeof(uw_horus_binary_v1); i++) {
+            hstates->uw[i] = 2*uw_horus_binary_v1[i] - 1;
+        }
+        hstates->uw_len = sizeof(uw_horus_binary_v1);
+        hstates->uw_thresh = sizeof(uw_horus_binary_v1) - 2; /* allow a few bit errors in UW detection */
         horus_l2_init();
         hstates->rx_bits_len = hstates->max_packet_len;
     }
-   
-    hstates->fsk = fsk_create(hstates->Fs, hstates->Rs, hstates->mFSK, 1000, 2*hstates->Rs);
+    // TODO: Horus 256/128-bit modes here.
+
+    // Create the FSK modedm struct. Note that the low-tone-frequency parameter is unused.
+    #define UNUSED 1000
+    hstates->fsk = fsk_create(hstates->Fs, hstates->Rs, hstates->mFSK, UNUSED, tx_tone_spacing);
+
```

Comment:
This is starting to look messy (also in fsk_modem.c). Think we need a better way of handling this case in the fsk.h API
+ maybe a FSK_UNUSED #define in fsk.h
+ a way to init the demod with an unused tone separation value when we are using the peak estimator.  The FSK modem API should trap the case of an FSK_UNUSED tone separation and mask being enabled.

I'll have a think about this one.

Reply:
Yes, I just used your approach from fsk_modem here...